### DISCUSSION TITLE
CS6800i/WLW176i - ECO+ Start Temperature unavailable (3.7.0-dev.23)

### DISCUSSION THREAD

**Discussion by MarkusMit** [emsesp/EMS-ESP32#1862] [Q&A] [answered]
After installing `3.7.0-dev.23 (ESP32-S3)` (#1852), I first thought everthing looks good, but yesterday I realized, that the field for "ECO+ Start Temperature" (`dhw/tempecoplus`) seems not to be working for our Bosch CS6800i heat pump.

![grafik](https://github.com/user-attachments/assets/1d08e80d-8fdb-4b8e-9167-ffaeaa7843fd)

**Comment by proddy** [comment]
Can you remember in which version it was working? Then we can trace the changes.

**Reply by MarkusMit** [reply]
Hi proddy, thanks for your prompt reply. I'm not sure it ever worked properly. 
I installed `3.7.0-dev.23` last Sunday because some properties were not working (see #1852). 
Before that I had `3.6.5` installed. (maybe I can try to step back tomorrow evening, but I cannot promise.)

**Reply by proddy** [reply]
We need to see if the value is there in a telegram, which is easy. You'll need to open Telnet to get to the console, then do a `read 8 ea`. Send the response back here. The `tempecoplus` should be in there somewhere. It could also be that it takes a minute to appear in the Web interface so keep checking too.

**Reply by MarkusMit** [reply]
Hi the output is this (still running on `3.7.0-dev.23`):
```
ems-esp-bosch:# read 8 ea
009+11:38:33.899 N 4: [emsesp] boiler(0x08) -W-> Me(0x0B), UBAParameterWWPlus(0xEA), data: 01 32 00 00 00 00 03 3C 00 01 18 3C 3C 2D (offset 5)
```
If my mind doesn't trick me, the value should be "43 °C" (`2B` hex).
Unfortunately I don't have physical access to the heat pump until friday night to check.

On the Web-Interface the value still doesn't show:
![grafik](https://github.com/user-attachments/assets/9bb40c67-b646-4232-a00a-0d83ace0b102)

**Reply by proddy** [reply]
I'll need Michael's help on this. I thought wwSelTempEcoplus would be at offset 27 in that 0xEA telegram, but yours is cut short. I think we may need to look at another telegram to find this data point.

**Reply by MarkusMit** [reply]
Thanks for looking into it! Let me know when I might be able to help.

**Reply by MichaelDvP** [reply]
Yes, we only have the values that are reported by users.  #1624 reports setpoint on 0xEA, offset 27.
The other values are in 0x499. 
0x499 still has a lot of unknown offsets. Check with `read 8 499` if your pump shows the setpoint there.

**Reply by MarkusMit** [reply]
0x499 contains this:
```
ems-esp-bosch:# read 8 499
010+11:42:47.701 N 5: [emsesp] boiler(0x08) -W-> Me(0x0B), HPDhwSettings(0x0499), data: 01 00 00 00 39 39 37 3C 0D 0B 09 (offset 4)
```
I don't see anything close to my (assumed) temperature of "43 °C" (`2B` hex). But I can check again on friday night, when I have again physical access to the device again.

**Reply by MarkusMit** [reply]
Hi again. Thanks to #CrowdStrike I'm home early today.
It looks like the eco+ start temperature is to find in neither of those two telegrams. 
I read the telegrams, then changed the temperature setting on the device, and read it again, but there was no change in the data:
```bash
## 0xEA # Eco Start Temperature: 43 °C
ems-esp-bosch:# read 8 ea
012+06:10:25.113 N 2: [emsesp] boiler(0x08) -W-> Me(0x0B), UBAParameterWWPlus(0xEA), data: 01 31 00 00 00 00 03 3C 00 01 18 3C 3C 2D (offset 5)
012+06:10:34.233 N 3: [emsesp] boiler(0x08) -W-> Me(0x0B), UBAParameterWWPlus(0xEA), data: 01 31 00 00 00 00 03 3C 00 01 18 3C 3C 2D (offset 5)
## 0xEA # Eco Start Temperature: 42 °C
ems-esp-bosch:# read 8 ea
012+06:10:56.732 N 4: [emsesp] boiler(0x08) -W-> Me(0x0B), UBAParameterWWPlus(0xEA), data: 01 31 00 00 00 00 03 3C 00 01 18 3C 3C 2D (offset 5)

## 0x499 # Eco Start Temperature: 43 °C
ems-esp-bosch:# read 8 499
012+06:11:21.921 N 5: [emsesp] boiler(0x08) -W-> Me(0x0B), HPDhwSettings(0x0499), data: 01 00 00 00 3A 39 37 3C 0D 0B 09 (offset 4)
## 0x499 # Eco Start Temperature: 42 °C
ems-esp-bosch:# read 8 499
012+06:11:44.505 N 6: [emsesp] boiler(0x08) -W-> Me(0x0B), HPDhwSettings(0x0499), data: 01 00 00 00 3A 39 37 3C 0D 0B 09 (offset 4)
```

**Reply by MichaelDvP** [reply]
Set the logger to log all and change the start temperatue on the device. We should see a telegram with a single byte changing send from thermostat to boiler.

**Reply by MarkusMit** [reply]
ah. never mind. I [found it](https://emsesp.github.io/docs/Console/).
<strike>Sorry, but I may need a bit more specific instructions in how to do that.
Do you mean the logging settings in the UI? And where to I find/monitor the logs then?</strike>

**Reply by MichaelDvP** [reply]
You can also use web interface, set log all, leave page open, got to device and change temperature ( a few times is better), got to computer and press the export button. 
![grafik](https://github.com/user-attachments/assets/4f26718f-9f24-4184-ae1d-681c65aa7c5b)

**Reply by MarkusMit** [reply]
Thanks for the hint.

But speaking of UI: I didn't check before but I just noticed now: Changing the setting on the device let the information **reappear in the UI** (and home assistant):
![ems-esp screenshot Temperatur settings](https://github.com/user-attachments/assets/062cb7b4-9016-45ca-8b94-d79e55d3c9c8)

So it seems it is only sent - in the `0xEA` telegram - when changed:
```bash
## setting to 42 on device
[...]
012+06:46:17.233 T 20: [emsesp] thermostat(0x10) -W-> boiler(0x08), UBAParameterWWPlus(0xEA), data: 2A (offset 27)
[...]
## setting to 43 on device
[...]
012+06:48:55.714 T 48: [emsesp] boiler(0x08) -B-> All(0x00), UBAParameterWWPlus(0xEA), data: 2B (offset 27)
[...]

## setting to 42 again via home assistant (now that the entity is available again):
012+06:58:33.773 I 59: [command] Calling command 'boiler/tempecoplus' (ECO+ Solltemperatur) with value 42 and id 9 on device 0x08
```

"Full" log of me changing settings:
[ems-esp-ecoplus-temp.log](https://github.com/user-attachments/files/16313942/ems-esp-ecoplus-temp.log)

**Reply by MichaelDvP** [reply]
Ah, the 0xEA have missing fields. `read 8 ea` gives only a small part of telegram. Try in terminal to read the higher offsets: `read 8 EA 1B` for offset 27 and later.
Your `boiler(0x08) -W-> Me(0x0B), UBAParameterWWPlus(0xEA), data: 01 31 00 00 00 00 03 3C 00 01 18 3C 3C 2D (offset 5)`
have offsets 5-18, Try the read command also with different values for offset. starting form offset 19 (0x13): `read 8 ea 13`
Maybe we find a rule to automatic fetch the higher offsets

**Reply by MarkusMit** [reply]
indeed, this way I see the value. 
also tried with `12`, it seems to bring the last value of the standard telegram and lots of zeros before the `2A`
```bash
ems-esp-bosch:# read 8 ea 13
012+07:27:55.349 N 60: [emsesp] boiler(0x08) -W-> Me(0x0B), UBAParameterWWPlus(0xEA), data: 2A (offset 27)

ems-esp-bosch:# read 8 ea 12
012+07:28:04.238 N 61: [emsesp] boiler(0x08) -W-> Me(0x0B), UBAParameterWWPlus(0xEA), data: 2D 00 00 00 00 00 00 00 00 2A (offset 18)
ems-esp-bosch:# read 8 ea
012+07:28:17.884 N 62: [emsesp] boiler(0x08) -W-> Me(0x0B), UBAParameterWWPlus(0xEA), data: 01 31 00 00 00 00 03 3C 00 01 18 3C 3C 2D (offset 5)
```

**Reply by MichaelDvP** [reply]
Do you get anything beyond offset 27? `read 8 ea 1c`, etc

**Reply by MarkusMit** [reply]
No, I only get `<empty>` data from `1c` onwards.

**Reply by MichaelDvP** [reply]
Ok. When fetching telegrams we read the next part only if the read is maximum telegram length. Just increment offset until we get an empty telegram back will give a lot of useless reads flooding the queue. 
I made an attempt to check the offset and for telegrams with holes (offset != requested offset) read the next parts..
Please update to https://github.com/MichaelDvP/EMS-ESP32/releases and make a full log of some minutes (> 5) to check the fetches.

**Reply by MarkusMit** [reply]
Thanks Michael. Here's the log of the new version for about the last five minutes:
[ems-esp-dev26-ALL-log.txt](https://github.com/user-attachments/files/16315182/ems-esp-dev26-ALL-log.txt)

**Reply by MichaelDvP** [reply]
Looks better, the starttemp should be there. But still a fix in offset is needed.
Uploaded a new version (also dev26), Please reflash and log again. I think this will work.
(My ems boiler only sends continious telegrams, can't test).

**Reply by MarkusMit** [reply]
no problem. :-) Here's another five minutes log. 
[ems-esp-dev26.2-ALL-log.txt](https://github.com/user-attachments/files/16315719/ems-esp-dev26.2-ALL-log.txt)

**Reply by MichaelDvP** [reply, marked as answer]
Looks ok now. There are some empty replies but that was expected. I have no other solution to get all values. 
But also the boiler requests some empty values, so Bosch also doesnt have another solution.

**Reply by MarkusMit** [reply]
Great. Thank you!
